### PR TITLE
WIP: update with `CTBase` 0.10, `CTDirect` 0.7, `CTFlows` 0.4 and `CTProblems` 0.7

### PR DESCRIPTION
**Pull request by ocots**
The CI is ok but we need to update all the tutorials.

### CODE REVIEW

**Comment by ocots** [comment]
Ci not ok. Need to remove tests with julia 1.9.

**Comment by jbcaillau** [comment]
https://github.com/control-toolbox/OptimalControl.jl/pull/185#issuecomment-2185923135

**Comment by PierreMartinon** [comment]
> Ci not ok. Need to remove tests with julia 1.9.

CTDirect has been requiring julia 1.10 for a few releases yes. The performance gains are such that I see no reason to support 1.9 unless someone specifically asks for it.

**Comment by jbcaillau** [comment]
> > Ci not ok. Need to remove tests with julia 1.9.
> 
> CTDirect has been requiring julia 1.10 for a few releases yes. The performance gains are such that I see no reason to support 1.9 unless someone specifically asks for it.

OK

@PierreMartinon Performance where? Loading / precompilation?

File: test/runtests.jl [hunk 1]
```diff
@@ -4,6 +4,7 @@ using LinearAlgebra
 using CTProblems
 using SciMLBase
 using NonlinearSolve
+using DifferentialEquations
```
Comment (jbcaillau) [inline test/runtests.jl:7]: @ocots nice! can't wait to check loading times without this `using` 🤞🏾

**Review by jbcaillau — APPROVED** [review]
@ocots @PierreMartinon `>= 1.10` it is

**Comment by ocots** [comment]
We should

> > The CI is ok but we need to update all the tutorials.
> 
> @ocots can you please elaborate a bit?

- [Basic example](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-basic-example.html): add `using Plots`, remove `size` kwarg in plot function.
- [Basic example functional](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-basic-example-f.html): add `using Plots`, remove `size` kwarg in plot function. Update the definition of the ocp from the new API.
- [Double integrator: time minimisation](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-double-integrator.html): add `using Plots`, remove `size` kwarg in plot function.
- [Initial guess](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-init.html): update the tutorial according to https://github.com/control-toolbox/OptimalControl.jl/issues/170.
- [Plot a solution](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-plot.html): add info about series attributes, labelling, time normalisation.
- [LQR example](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-lqr-basic.html): add `using Plots`. Use the time normalisation within the plot function using `time=:normalized` instead of defining a rescaling function.
- [Indirect simple shooting](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-iss.html): add `using DifferentialEquations`, present `MINPACK` and `NonlinearSolver` (`MINPACK` cannot be used with arm).
- [Batch processing](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-batch.html): add `using Plots`, remove `size` kwarg in plot function. Add a label for the two plots using `solution_label` kwarg.
- [Goddard problem](https://control-toolbox.org/docs/optimalcontrol/dev/tutorial-goddard.html): add `using Plots`, remove `size` kwarg in plot function. Add a label for the two plots (`direct` and `indirect`) using `solution_label` kwarg. Add `using DifferentialEquations`.

**Comment by PierreMartinon** [comment]
> > > Ci not ok. Need to remove tests with julia 1.9.
> > 
> > 
> > CTDirect has been requiring julia 1.10 for a few releases yes. The performance gains are such that I see no reason to support 1.9 unless someone specifically asks for it.
> 
> OK
> 
> @PierreMartinon Performance where? Loading / precompilation?

I assume so. This was the time when new warnings appeared during precompilation.

**Pull request merged**